peps/pep-0752.rst

```diff
@@ -0,0 +1,399 @@
+
+* `NPM <https://www.npmjs.com>`__ has the concept of
+  `scoped packages <https://docs.npmjs.com/about-scopes>`__ which were
+  `introduced`__ primarily to combat there being a dearth of available good
+  package names (whether a real or perceived phenomenon). When a user or
+  organization signs up they are given a scope that matches their name. For
+  example, the
+  `package <https://www.npmjs.com/package/@google-cloud/storage>`__ for using
+  Google Cloud Storage is ``@google-cloud/storage`` where ``@google-cloud/`` is
+  the scope. Regular user accounts (non-organization) may publish `unscoped`__
+  packages for public use.
+  This approach has the lowest amount of backwards compatibility because every
+  installer and tool has to be modified to account for scopes.
+* `NuGet <https://www.nuget.org>`__ has the concept of
+  `package ID prefix reservation`__ which was
+  `introduced`__ primarily to satisfy users wishing to know where a package
+  came from. A package name prefix may be reserved for use by one or more
+  owners. Every reserved package has a special indication
+  `on its page <https://www.nuget.org/packages/Google.Cloud.Storage.V1>`__ to
+  communicate this. After reservation, any upload with a reserved prefix will
+  fail if the user is not an owner of the prefix. Existing packages that have a
+  prefix that is owned may continue to release as usual. This approach has the
+  highest amount of backwards compatibility because only modifications to
+  indices like PyPI are required and installers do not need to change.
+
+__ https://blog.npmjs.org/post/116936804365/solving-npms-hard-problem-naming-packages
+__ https://docs.npmjs.com/package-scope-access-level-and-visibility
+__ https://learn.microsoft.com/en-us/nuget/nuget-org/id-prefix-reservation
+__ https://devblogs.microsoft.com/nuget/Package-identity-and-trust/
+
+This PEP specifies the NuGet approach of authorized reservation across a flat
+namespace for the following reasons:
+
+* Causing churn for the community is a hard blocker.
+* The NPM approach has the potential to cause confusion for users if we allow
+  unscoped names. Our community has chosen to normalize separator characters
+  and so ``@aws/s3`` would likely be confused with ``@aws-s3``.
+
+Granting Namespaces
+-------------------
+
+PyPI has been understaffed, receiving the first `dedicated specialist`__ in
+July 2024. Due to lack of resources, user support has been lacking for
+`package name claims <https://discuss.python.org/t/27436/19>`__,
+`organization requests <https://discuss.python.org/t/33764/15>`__,
+`storage limit increases <https://discuss.python.org/t/54035>`__,
+and even `account recovery <https://discuss.python.org/t/43422/122>`__.
+
+__ https://pyfound.blogspot.com/2024/07/announcing-our-new-pypi-support.html
+
+Reserving the ability to claim prefixes for
+`corporate organizations <corp-orgs_>`_ provides the following benefits:
```

Comment:
If namespaces are a funding source, we have definitely get feedback from the PSF to see what is even legal.  If corporations can pay for orgs, would that include namespace benefits or would namespace benefits be a payment add-on?  Do you anticipate orgs that wouldn't want to claim namespaces if they could?  Can non-profits and loose collectives of OSS aficionados participate?

Reply:
I did speak to Deb and Loren, billing is now set up properly as directed by legal. I don't know exactly what that means but this is now possible 🙂